applications/mne_anonymize/mainwindow.cpp

```diff
@@ -0,0 +1,925 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ * @brief    Mainwindow class definition.
+ *
+ */
+
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "mainwindow.h"
+#include "ui_mainwindow.h"
+#include <settingscontrollergui.h>
+#include <iostream>
+
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QMessageBox>
+#include <QCloseEvent>
+#include <QUrl>
+#include <QDesktopServices>
+#include <QDateTime>
+#include <QFileDialog>
+
+//=============================================================================================================
+// EIGEN INCLUDES
+//=============================================================================================================
+
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace MNEANONYMIZE;
+
+//=============================================================================================================
+// DEFINE GLOBAL METHODS
+//=============================================================================================================
+
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+MainWindow::MainWindow()
+: m_bOptionsVisibility(false)
+, m_iDefaultWindowHeight(222)
+, m_iDefaultWindowHeightLarge(666)
+, m_bShowWraningMsgBoxInWasm(true)
+, m_sDefaultWasmInFile("/in.fif")
+, m_sDefaultWasmOutFile("/out.fif")
+, m_pUi(new Ui::MainWindow)
+{
+    m_pUi->setupUi(this);
+    setDefautlStateUi();
+    setDefaultStateExtraInfo();
+    setupConnections();
+}
+
+//=============================================================================================================
+
+MainWindow::~MainWindow()
+{
+    delete m_pUi;
+}
+
+//=============================================================================================================
+
+void MainWindow::closeEvent(QCloseEvent *event)
+{
+    if(confirmClose())
+    {
+        event->accept();
+    } else {
+        event->ignore();
+    }
+}
+
+//=============================================================================================================
+
+bool MainWindow::confirmClose()
+{
+    return true;
+//    const QMessageBox::StandardButton ret
```

Comment:
do we still need this switch, otherwise remove

Reply:
At some point there could be some interest in: 

 - be able to modify each and every field.
 - if so, it could be interesting to save values in a text file that could be loaded back. If modifications are made, I want to be warned to save them before exiting. Code is just a reminder.

Reply:
ok in that case we should keep it